carbon-design-system/carbon-addons-iot-react
# fix(Hierarchylist): style, i18n, selection

**Summary**

- Merge i18n prop with the default props to ensure all get passed through
- Add `hasDeselection` prop to remove the ability to deselect an item once its selected
  - When `hasDeselection` is false, don't allow deselection and don't refire `onSelect`
- Padding on sides should be 16px instead of 8
- Padding should be reduced when an item is select to account for the selected border

**Change List (commits, features, bugs, etc)**

- Added tests and story knobs for new prop
- Added knob for search i18n

**Acceptance Test (how to verify the PR)**

- Go to /?path=/story/watson-iot-hierarchylist--stateful-list-with-nested-searching , select an item, then try to select it again. It should not get deselected and it should not fire `onSelect`
- Go to the same story and attempt to edit the i18n search prop. You should see the changes live in the search bar
